LICENSE.md

```diff
@@ -67,24 +70,37 @@ The user has the right to decompile the binaries and examine the resulting sourc
+- The distributor must attribute the author of the mod near any download buttons or other calls to acquire the mod by clearly displaying the name of the author and/or the URL of the author's website.
+- Unmodified copies may only be distributed through the link the mod author provides. However, the original mod author may grant access to a distributor to use a link other than the one which they provide.
+- Only the mod author can distribute this mod in a way which generates any form of revenue. However, an original mod author may grant permission to a distributor to distribute this mod in a way which generates a form of revenue
+- The original mod author may revoke hosting permission from any distributor.
 
 6. Right to distribute derivatives
 ----------------------------------
 
 Derivatives of this mod include, but are not limited to:
-   - modified versions of the mod source code or binaries (including binaries compiled from the modified source code)
-   - any portion of the binary or source files
+- modified versions of the mod source code or binaries (including binaries compiled from the modified source code)
+- any portion of the binary or source files
 
 The user is granted the right to distribute derivatives with the following restrictions:
-   - The derivative must carry prominent notices that you modified the mod.
-   - The derivative must be distributed under the same license as this mod.
-   - The entire source code of the derivative must made freely available without charge or fee.
+- The derivative must carry prominent notices that you modified the mod.
+- The derivative must be distributed under the same license as this mod.
+- The entire source code of the derivative must made freely available without charge or fee.
+- The derivative may not be used to generate revenue. However, the original mod author may allow the derivative be used to generate revenue
 
 7. Right to distribute addons
 -----------------------------
 
-Ths user is granted the right to distribute addons to this mod, which may be distributed under a different license, as long as the addon is not a derivative or it includes only the API of this mod. For example, if an addon extended this mod using the mod's API, but did not include other derived code, then it can be distributed under another license.
+The user is granted the right to distribute addons to this mod, which may be distributed under a different license, as long as the addon is not a derivative or it includes only the API of this mod. For example, if an addon extended this mod using the mod's API, but did not include other derived code, then it can be distributed under another license.
+
+8. Right to distribute in a modpack
+-----------------------------------
+
+Using the link the author provides is not required when distributing mods in a modpack. However, modpacks may only be hosted on sites/launchers given permission by the original mod author. Modpacks may not be used to generate revenue.
```

Comment:
I kinda want to change this a bit...I'd prefer that the license grant modpack permissions by default, but give the author the ability to revoke permissions on a case by case basis.